### ISSUE TITLE
Issues with Solar Auxilia HQs

### ISSUE THREAD

**Issue by TomTheHand** [BSData/horus-heresy#1120]
I've looked over the Solar Auxilia HQs and found some issues; I haven't looked at the rest of the Auxilia in this level of detail yet.

**Legate Commander (Crusade Imperialis, pg. 24)**
- [x] Hand Flamer, Inferno Pistol, Plasma Pistol, and Power Fist do not point to weapon profiles.
- [x] Currently it is not possible to select a specific type of Power Weapon.
- [x] The Volkite Serpenta does not cause the Deflagrate special rule to display (it does correctly list Deflagrate in its type).
- [x] Selecting an Iron Halo or Displacer Matrix does not remove the Refractor Field entry from the unit profile.
- [x] Selecting Digital Lasers does not add a descriptive entry to the unit profile.
- [x] Selecting Artificer Armour does not modify the Save in the stat line, add a descriptive entry to the unit profile, or remove the entry for Void Armour.
- [x] Perhaps upgrading a Legate Commander to Lord Marshal should change the unit name.

**Lord Mashal Ireton MaSade (Crusade Imperialis, pg. 25)**
- [x] MaSade's profile does not include his master-crafted Power Sword.
- [x] MaSade should be restricted to Loyalist allegiance.

**Auxilia Tactical Command Section (Crusade Imperialis, pg. 26)**
- [x] For the Strategos, the Hand Flamer, Plasma Pistol, and Power Fist do not point to profiles.
- [x] Currently it is not possible to select a specific type of Power Weapon for the Strategos.
- [x] Veteran Auxiliaries have individual entries for each model, and currently have their lasrifles implemented as a numeric entry that defaults to 1. Validation errors are thrown correctly if the user increases the number above 1, if the user selects a special weapon without setting it to 0, or if the user sets it to 0 without selecting a special weapon, but this is still a really wonky implementation.
- [x] For Veteran Auxiliaries, the Flamer, Meltagun, and Plasma gun do not point to profiles.
- [x] For Veteran Auxiliaries, the Volkite Charger does not cause the Deflagrate special rule to display (it does correctly list Deflagrate in its type).
  _Note that Veletaris Storm Sections and Household Retinue Squads, which also use Volkite Chargers, DO display the Deflagrate special rule)._
- [x] The Command Section should be able to be used as the Compulsory HQ.

**Auxilia Tank Commander (Crusade Imperialis, pg. 27)**

- [x] The user should be required to take an appropriate tank for each commander.
  _Note that the way the rule is phrased indicates that the tank is added separately and would therefore still take up its usual slot and could potentially still be part of a unit (such as a squadron of Leman Russes). I would think this means the commander's tank can't be implemented in the same way as a Legion Master of Armour._

**Comment by capitaladot** [comment]
Another good issue submission, thank you.

On the Tank Commander, Categories make this possible.

**Comment by zopha** [comment]
I've separated out the special weapons Auxilleries so they're added individually and you select the weapon for each of them.

<img width="579" alt="2017-10-24 11_08_20-roster editor v2 01 05 - new roster warhammer 30 000 - the horus heresy v89" src="https://user-images.githubusercontent.com/1666415/31937215-adce3dc6-b8ab-11e7-8dcf-421902ad6b36.png">

The rest has been updated. I have some ideas I need to test for the tank commander and I'll work on that over the next few days.
